src/runtime/test_case_runner.js

```diff
@@ -182,6 +200,25 @@ export default class TestCaseRunner {
     })
   }
 
+  async runStepHooks(step, hookDefinitions) {
+    const stepHookResults = await Promise.all(
+      hookDefinitions.map(async hookDefinition =>
+        this.runStepHook(step, hookDefinition)
+      )
+    )
+    const errors = stepHookResults.filter(result => result.exception)
+    if (errors && errors.length) {
+      return {
+        exception: 'Errors running step hooks',
+        status: Status.FAILED,
+      }
+    } else {
+      return {
+        status: Status.PASSED,
+      }
+    }
+  }
```

Comment:
Thinking on this function, we are currently not adding the duration of the step hooks to the overall duration of the step. That should be testable with unit tests